extensions/todoist/src/menubar.tsx

```diff
@@ -9,13 +9,19 @@ import { Task } from "@doist/todoist-api-typescript";
 
 export default function Command() {
   const isTodayView = getPreferenceValues().view === "today";
+  const upcomingDays = getPreferenceValues().upcomingDays;
+  const isUpcomingDaysView = upcomingDays !== "";
 
   const {
     data: tasks,
     isLoading: isLoadingTasks,
     error: tasksError,
     mutate: mutateTasks,
-  } = useCachedPromise(() => todoist.getTasks({ filter: isTodayView ? "today|overdue" : "all" }));
+  } = useCachedPromise(() =>
+    todoist.getTasks({
+      filter: isTodayView ? "today|overdue" : isUpcomingDaysView ? `due before: +${upcomingDays} days` : "all",
```

Comment:
Let's make sure beforehand that `upcomingDays` is a number (`isNaN(Number(upcomingDays))` should be enough). If a user makes a mistake in the preference, the view may be messed up.